platform/vcs-impl/src/com/intellij/openapi/vcs/actions/VcsContextWrapper.java

```diff
@@ -160,7 +160,7 @@ public Refreshable getRefreshableDialog() {
       stream(VcsDataKeys.FILE_PATH_ARRAY.getData(myContext)),
       getSelectedFilesStream().map(VcsUtil::getFilePath),
       stream(getSelectedIOFiles()).map(VcsUtil::getFilePath)
-    );
+    ).distinct();
```

Comment:
This will make the method actually process all elements, i.e. make it non-lazy. It would eliminate the performance optimization that have been made.
CC @pchel-